etc/inc/disable-common.inc

```diff
@@ -394,18 +394,24 @@ blacklist /tmp/ssh-*
 # top secret
 blacklist /.fscrypt
 blacklist /etc/davfs2/secrets
+blacklist /etc/doas.conf
 blacklist /etc/group+
 blacklist /etc/group-
 blacklist /etc/gshadow
 blacklist /etc/gshadow+
 blacklist /etc/gshadow-
+blacklist /etc/pam.d
```

Comment:
Will break sshd and maybe some other

Reply:
@rusty-snake [on Oct 2](https://github.com/netblue30/firejail/pull/5387#discussion_r985139410):

> Will break sshd and maybe some other

There is no profile for sshd.

I don't know much about PAM, but wouldn't this only affect login managers,
system daemons (which usually don't have profiles) or things that run outside
of firejail anyway?

Reply:
Some distros/packages/plugins/patches have a pam.d/sshd AFAIK. But I need to check later what else is in there.

Reply:
Uh, we don't have a sshd profile. I thought we have one.

Reply:
@rusty-snake [on Oct 2](https://github.com/netblue30/firejail/pull/5387#discussion_r985197303):

> Some distros/packages/plugins/patches have a pam.d/sshd AFAIK. But I need to
> check later what else is in there.

To be clear, I meant that there is no firejail profile for sshd.

From pam(8):

> Linux-PAM is a system of libraries that handle the authentication tasks of
> applications (services) on the system. The library provides a stable general
> interface (Application Programming Interface - API) that privilege granting
> programs (such as login(1) and su(1)) defer to to perform standard
> authentication tasks.

Is there any unprivileged application that uses PAM directly?

I think only system package managers would need to write to /etc/pam.d, and
they are already privileged anyway (and there are no firejail profiles for them
AFAIK).

Reply:
We never had a profile for sshd, only for ssh and ssh-agent. IMO that's all we should do, the sshd daemon is better left protected via systemd.

Reply:
> To be clear, I meant that there is no firejail profile for sshd.

Yes, I thought we had one.

> Is there any unprivileged application that uses PAM directly?

```console
$ ls /etc/pam.d 
atd          fingerprint-auth@       gdm-password   passwd          ppp        smartcard-auth@   sudo          vlock
config-util  gdm-autologin           gdm-smartcard  password-auth@  remote     sshd              sudo-i        xserver
crond        gdm-fingerprint         login          polkit-1        runuser    sssd-shadowutils  su-l
cups         gdm-launch-environment  other          postlogin@      runuser-l  su                system-auth@
```

Problems that I see are with awesome, blackbox, fluxbox, openbox, sway, tmux, ... since they can contain a (terminal)-session using su(do) or a screen-locker.